components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1223,4 +1305,202 @@ private static boolean isSelfSignedCert(X509Certificate cert) {
         }
     }
 
+    private static void addDefaultValidatorConfigToConfigStore(OMElement validatorsElement, String tenantDomain)
+            throws CertificateValidationException {
+        List<Validator> defaultValidatorConfig = getDefaultValidatorConfig(validatorsElement);
+
+        for (Validator validator : defaultValidatorConfig) {
+            String validatorResourceType = X509CertificateValidationConstants.VALIDATOR_RESOURCE_TYPE;
+            if (log.isDebugEnabled()) {
+                log.debug("Adding default validator configurations to config store in: " + validatorResourceType);
+            }
+            try {
+                CertValidationDataHolder.getInstance().getConfigurationManager().getResourceType(validatorResourceType);
+            } catch (ConfigurationManagementException e) {
+                addResourceTypeIfNotExists(e, validatorResourceType, tenantDomain);
+            }
+            try {
+                addValidatorConfigInRegistry(validatorResourceType, validator);
+            } catch (ConfigurationManagementException e) {
+                if (ERROR_CODE_RESOURCE_ALREADY_EXISTS.getCode().equals(e.getErrorCode())) {
+                    log.debug("Resource already exists in the tenant config store.");
+                } else {
+                    throw new CertificateValidationException("Error while adding validator configurations.", e);
+                }
+            }
+        }
+    }
+
+    private static void addResourceTypeIfNotExists(ConfigurationManagementException e, String resourceType,
+                                                   String tenantDomain) throws CertificateValidationException {
+        if (ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
+            ResourceTypeAdd resourceTypeAdd = new ResourceTypeAdd();
+            resourceTypeAdd.setName(resourceType);
+            try {
+                CertValidationDataHolder.getInstance().getConfigurationManager().addResourceType(resourceTypeAdd);
+            } catch (ConfigurationManagementException ex) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Error while adding resource type: " + resourceType, ex);
+                }
+            }
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Configuration for %s is added to %s tenant config store.",
+                        resourceType, tenantDomain));
+            }
+        } else if (!ERROR_CODE_RESOURCE_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
```

Comment:
Do we need this check?
ie. would `getResourceType()` method  throws an exceptions with `ERROR_CODE_RESOURCE_DOES_NOT_EXISTS` error code and why we have to ignore such exceptions?

Reply:
```
CertValidationDataHolder.getInstance()
                        .getConfigurationManager().getResource(X509_CA_CERT_ALIAS, CERTS);
```

When adding resources also they inherantly called getReosurceTypes and whether resource type is exist. And if resourceType exist they check whether resource exists.

Reply:
Still its not clear how this error could occur.. Are you saying this error comes when `addResource()` get called?But here we are handling exceptions for `getResourceType()` call.

Reply:
removed getResourceType since we are adding the types through db scripts